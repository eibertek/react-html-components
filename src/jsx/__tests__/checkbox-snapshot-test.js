import React from 'react';
import Checkbox from '../checkbox';
import renderer from 'react/lib/ReactTestRenderer';

const checkboxWithListeners = (
    <Checkbox required={true}
                 label="test label"
                 name="test_input"
                 id="test"
                 placeholder="Checkbox"
                 extraClass="anotherClass"
                 validate={true}
                 errorMessage="wrong"
                 successMessage="correct"
                 changeCallback={function(){}}
                 mouseEnterCallback={function(){}}
                 mouseLeaveCallback={function(){}}/>
);

const checkboxWithoutListeners = (
    <Checkbox required={true}
                 label="test label"
                 name="test_input"
                 checked={true}
              filledIn={true}
                 id="test"/>
);

const checkboxWithValue = (
    <Checkbox label="test label"
                 name="test_input"
                 id="test"
                 checked={false}
                 value="value"/>
);

describe('Checkbox', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(checkboxWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(checkboxWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(checkboxWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
