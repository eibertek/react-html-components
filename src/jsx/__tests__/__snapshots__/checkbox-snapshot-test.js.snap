exports[`Checkbox with listeners matches snapshot 1`] = `
<p>
  <input
    className="validate anotherClass"
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder="Checkbox"
    required={true}
    type="checkbox" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error="wrong"
    data-success="correct"
    htmlFor="test" />
</p>
`;

exports[`Checkbox with value matches snapshot 1`] = `
<p>
  <input
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={false}
    type="checkbox" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</p>
`;

exports[`Checkbox without listeners matches snapshot 1`] = `
<p>
  <input
    className=" filled-in"
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="checkbox" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</p>
`;
